### PR TITLE
Local bitvector analysis: use parameter_identifiers [blocks: #4167]

### PR DESCRIPTION
**Pull request by tautschnig**
goto_functiont::type will go away.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by kroening — APPROVED** [review]

**Review by smowton — APPROVED** [review]
Suggestion for the future: when editing old code like this much of the change is just modernisation of for-loops, style improvement and so on. At least for review, how about two commits, one purely style and one that makes the real change?

File: src/analyses/local_bitvector_analysis.cpp [hunk 1]
```diff
@@ -254,9 +250,9 @@ void local_bitvector_analysist::build()
   // Gather the objects we track, and
   // feed in sufficiently bad defaults for their value
   // in the entry location.
-  for(const auto &local : locals.locals_map)
-    if(is_tracked(local.first))
-      loc_infos[0][pointers.number(local.first)]=flagst::mk_unknown();
+  for(const auto &local : locals.locals)
```
Comment (thk123) [inline src/analyses/local_bitvector_analysis.cpp:253]: :pick: Body of for is spread over more than one line, could it be wraped in braces

**Review by thk123 — APPROVED** [review]

File: src/analyses/local_bitvector_analysis.cpp (hunk 1 above)
Comment (tautschnig) [reply src/analyses/local_bitvector_analysis.cpp:253]: ✔️

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 92e41bcb9161d74e809cc68a1bfd6c7032d0119b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102435916

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 419382999bcb37730423ef2f5fb0f8b017fe44f3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102908293

**Pull request merged**